### PR TITLE
Made OOP.  Added Annotate class containing distinct classes to automa…

### PR DESCRIPTION
**Pull request by markhankins**
…tically add all pods which have a PVC on a PV.

### CODE REVIEW

**Comment by zakkg3** [comment]
Hey I missed this notification. I'll take a look in the coming days and review / lgtm / merge 
Thanks for the PR!

File: README.md [hunk 1]
```diff
@@ -1,35 +1,89 @@
 # podannotator
-Annotate pods with PVC in a given list of namepsaces to be backed up by velero/restic
+Annotates all volumes (with a PVC), in all pods, in a given list of namepsaces to be backed up by velero/restic.
 
+Namespace 1
```
Comment (zakkg3) [inline README.md]: Please take a look on how the output looks like. https://github.com/markhankins/podannotator/tree/v2
You may want to ad 3 quotes simbols to preserve the spacing:

```
        -- Pod 2
                -- Volume 1 (N1P2V1)
                -- Volume 2 (N1P2V2)
```

File: README.md [hunk 2]
```diff
@@ -1,35 +1,89 @@
...
-It will annotate all the pods with persist volume claims with the anotation: `backup.velero.io/backup-volumes: volume-name-to-backup`
-So now you can backup the namespace with velero/restic and this pvc will be backed up aswell. :)
 
+## Initial position display
 
-# Output:
+Initially, a table will be created showing the start position of all of the volumes within all of the pods within the provided namespace(s).
 
+For example, if I issued the following command: python3 annotate.py raffia,logstash,vault,nagios
+
```
Comment (zakkg3) [inline README.md:29]: same as comment # 1

File: README.md [hunk 3]
```diff
@@ -1,35 +1,89 @@
...
+The reason for re-mapping the return from the API call is that this makes life much simpler later when we want to tie all volumes to their respective parent pods, otherwise we need to keep looping in and out of the API call repeatedly, which is more requests, and all takes additional time.  If this needs to be extended to include more data, this can be done by extending the get_pod_map method within the Annotate class.
+
+
+## Human readable preview
+
+A human-readable preview will be shown next.  This is a legacy from a prompt that was shown (but this would cause issues with pipelining the script) so has been removed for now (Annotate.py line 46-48).  It could be added back in later if required, but for now, it was decided to leave the human-readable list in place for simpler review during pipelining.
+
+
+## Final position display
+
```
Comment (zakkg3) [inline README.md:78]: and again same as comment # 1 
:)

File: README.md [hunk 4]
```diff
@@ -1,35 +1,89 @@
...
+                -- Volume 2 (N1P2V2)
+Namespace 2
+        -- Etc.
+
+Note: All pods within a given namespace will be backed up, if they have a PVC.  (Ignoring PVC's would include secrets.)  There is currently no requirement to be able to exclude volumes from backups.  See @todo in class if this is ever in scope.
 
 # Usage  
 
-        annotator namespace1 namespace2
+python3 annotator.py namespace1,namespace2
```
Comment (zakkg3) [inline README.md:19]: can you add the requierment 
`pip3 install prettytable`
thanks :)

**Review by zakkg3 — CHANGES_REQUESTED** [review]
A part of this formatting issues on the readme, is tested on docker-desktop and working like a charm :)
/lgtm

File: README.md (hunk 1 above)
Comment (markhankins) [reply README.md]: Have changed it to indented lists.

File: README.md (hunk 4 above)
Comment (markhankins) [reply README.md:19]: Sorry, I don't understand where you mean?

File: README.md (hunk 2 above)
Comment (markhankins) [reply README.md:29]: Table fixed.

File: README.md (hunk 3 above)
Comment (markhankins) [reply README.md:78]: Table fixed.

**Review by markhankins — COMMENTED** [review]
Have fixed the formatting issues.  Only one comment that I wasn't sure about around including prettytable?

File: README.md (hunk 4 above)
Comment (markhankins) [reply README.md:19]: @zakkg3 Any chance you could review this for me please so we can get it merged (or fixed if more fixes are required?)

Thanks :)

**Comment by zakkg3** [comment]
> prettytable

its a requirement we may want to include in the docs.

**Pull request merged**